doc/manual/building_developing.rst

```diff
@@ -0,0 +1,172 @@
+
+Nix development environment
+---------------------------
+
+* Install `Nix <http://nixos.org/nix/>`_ if you haven't already, version 2.4 or later. Prefer a single-user installation for simplicity. 
+* Enable flakes in Nix, for example by adding ``experimental-features = nix-command flakes`` to ``nix.conf``; see the `NixOS Wiki on flakes <https://nixos.wiki/wiki/flakes>`_ for details and more options. 
+* Clone `the ARTIQ Git repository <https://github.com/m-labs/artiq>`_, or `the ARTIQ-Zynq repository <https://git.m-labs.hk/M-Labs/artiq-zynq>`__ for Zynq devices (Kasli-SoC or ZC706). By default, you are working with the ``master`` branch, which represents the beta version and is not stable (see :doc:`releases`). Checkout the most recent release (``git checkout release-[number]``) for a stable version. 
+* If your Vivado installation is not in its default location ``opt``, open ``flake.nix`` and edit it accordingly (once again text-search ``/opt/Xilinx/Vivado``). 
+* Run ``nix develop`` at the root of the repository, where ``flake.nix`` is. 
+* Answer ``y``/'yes' to any Nix configuration questions if necessary, as in :ref:`installing-troubleshooting`. 
+
+.. note:: 
+    You can also target legacy versions of ARTIQ; use git to checkout older release branches. Note however that older releases of ARTIQ required different processes for developing and building, which are broadly more likely to figure out by (also) consulting corresponding older versions of the manual.  
+
+Once you have run ``nix develop`` you are officially in the ARTIQ development environment. All ARTIQ commands and utilities -- ``artiq_run``, ``artiq_master``, etc. -- should be available, as well as all the packages necessary to build or run ARTIQ itself. You can exit the environment at any time using Control+D or the ``exit`` command and re-enter it by re-running ``nix develop`` again in the same location. 
+
+In original ARTIQ, if you only intend to rebuild, without changing the source code, you may also use the ARTIQ flake's provided ``artiq-boards-shell``, a lighter environment optimized for building firmware and flashing boards. You can do this by running ``nix develop .#boards`` instead. Developers should note that in this shell the current copy of ARTIQ sources is not added to your ``PYTHONPATH``. Run ``nix flake show`` and read ``flake.nix`` carefully to understand the different available shells. 
+
+Building ARTIQ
+--------------
+
+For general troubleshooting and debugging, especially with a 'fresh' board, see also :ref:`connecting-uart`. 
+
+Kasli or KC705 (ARTIQ original) 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you have your system description JSON file on-hand, you can at this point build the firmware and gateware with a command such as: :: 
+
+    $ python -m artiq.gateware.targets.kasli <description>.json
+    $ python -m artiq.gateware.targets.kc705 <description>.json
```

Comment:
kc705 does not use JSON descriptor files, but rather hardcoded variants (``nist_qc2``, ``nist_clock`` etc.)

Reply:
:thinking: